x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -90,12 +91,26 @@ export const hasReadIndexPrivileges = async (args: {
   logger: Logger;
   buildRuleMessage: BuildRuleMessage;
   ruleExecutionLogger: IRuleExecutionLogForExecutors;
+  savedObjectsClient: SavedObjectsClientContract;
+  version: string;
 }): Promise<boolean> => {
-  const { privileges, logger, buildRuleMessage, ruleExecutionLogger } = args;
+  const { privileges, logger, buildRuleMessage, ruleExecutionLogger, savedObjectsClient, version } =
+    args;
+  const config = await withSecuritySpan('getCcsWarningEnabled', () =>
+    savedObjectsClient.get<{ 'securitySolution:enableCcsWarning': boolean }>('config', version)
+  );
```

Comment:
We don't have access to the `uiSettingsClient` at setup time which is when the rules are defined. The alerting team gives us the `savedObjectClient` by passing it in as a parameter to the rule executors. They could probably do that with the `uiSettings` but it would require changes to the alerting framework that we need their input on

Reply:
cc @lukeelmers did you see that during your review? Are we fine with it?

Reply:
@lukeelmers @pgayvallet We worked with the alerting team to pass the `uiSettingsClient` through the services object so we're able to access the ui settings directly, would y'all mind taking another look?